lib/componentRelay.ts

```diff
@@ -256,16 +256,16 @@ export default class ComponentRelay {
         const originalMessage = this.sentMessages?.filter((message: MessagePayload) => {
           return message.messageId === payload.original?.messageId
         })[0]
-  
-        if (!originalMessage) {
-          // Connection must have been reset. We should alert the user.
+
+        if (!originalMessage && payload.action !== ComponentAction.Reply) {
+          // Connection must have been reset. We should alert the user unless it's a reply,
+          // in which case we may have been deallocated and reinitialized and lost the
+          // original message
           Logger.error('This extension is attempting to communicate with Standard Notes, but an error is preventing it from doing so. Please restart this extension and try again.')
```

Comment:
Hmm... the above line only outputs a message to console, and shouldn't be showing an alert. Perhaps the alert you are seeing is from `snjs`: https://github.com/standardnotes/snjs/blob/44217a60a0a7d1907c10da27177a4ce1979d72f6/packages/snjs/lib/services/component_manager.ts#L736?

Reply:
Ohh maybe it's because I was testing using the Bold Editor and this hasn't been updated with the new component-relay? Because the alert I was seeing matched this text 100%.

Reply:
Legacy: https://github.com/standardnotes/component-relay/blob/dd883886e2cc20c5aebcfc7c9467bbd46ca5a503/src/componentManager.js#L97

I wonder if it's dangerous not to alert the user, because the purpose for the alert was originally because if this state happened it means the user is typing but their changes could be not being saved.

Reply:
Yeah, makes sense to add this back.

Btw, what if we include the extension name in the alert message? Because "This extension is attempting..." does not specify which extension is having issues. It's easy to figure out this when you just have one extension activated, but when you have more than one, it can become tricky. 

We could set the extension name by passing it to the `ComponentRelay` constructor.

Reply:
Yeah adding the name would be great, but the relay doesn't have access to it. SNJS would have to pass it to the editor, because the user may have renamed the editor to something else, so it can't be hardcoded into the editor code. Would have to come from component.package_info.name on the SNJS side. We can definitely add that.

Reply:
Maybe we can also use document.title? This wouldn't have the correct name if a user renamed the editor, but maybe at least it's a better hint than nothing, and won't require SNJS changes.

Reply:
We can use `document.title`, but I like more getting this from SNJS, from `component.packge_info.name`. Perhaps we could go with `document.title` and later implement this in another PR.

Reply:
Sounds good, feel free to take over from here and merge/release this PR when you think it's ready. I'm now much less familiar with this repo than you are 😄